frontend/src/metabase/databases/components/DatabaseForm/DatabaseForm.unit.spec.tsx

```diff
@@ -0,0 +1,137 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import type { Engine } from "metabase-types/api";
+
+import DatabaseForm from "./DatabaseForm";
+
+interface SetupOpts {
+  onSubmit?: () => void;
+}
+
+function setup({ onSubmit }: SetupOpts) {
+  render(<DatabaseForm engines={engines} isAdvanced onSubmit={onSubmit} />);
+}
+
+describe("DatabaseForm", () => {
+  it("should submit default values", async () => {
+    const mockOnSubmit = jest.fn();
+    setup({ onSubmit: mockOnSubmit });
+
+    const expectedDatabaseName = "My H2 Database";
+    const expectedConnectionString = "file:/somewhere";
+    userEvent.type(screen.getByLabelText("Display name"), expectedDatabaseName);
+    userEvent.type(
+      screen.getByLabelText("Connection String"),
+      expectedConnectionString,
+    );
+
+    const saveButton = screen.getByRole("button", { name: "Save" });
+    await waitFor(() => {
+      expect(saveButton).toBeEnabled();
+    });
+    userEvent.click(saveButton);
+
+    const expectedDefaultSchema = {
```

Comment:
why isn't it an outside constant as `engines`?

Reply:
Good point, I can move this outside